### ISSUE TITLE
Tor disabled at install

### ISSUE THREAD

**Issue by jlj2** [subgraph/subgraph-os-issues#176]
When starting up Tor browser immediately after installing Subgraph, the Tor button next to the address bar indicated "Tor disabled".  This has happened since install.

Can browse the internet using Tor.  However, the 'Onion' applet in the Gnome top bar is shaded out.

To obtain report data, when selecting  "Tor Network Settings", a "Tor Launcher" message box reported "Unable to retrieve Tor settings.  Please ensure that Tor is running." However, in the background a "Tor Network Settings" dialog box appears:

![screenshot from 2016-08-29 17-26-29](https://cloud.githubusercontent.com/assets/17560937/18113662/13486620-6f21-11e6-8de0-6b69bb71a2a2.png)

In case it can be of help - when choosing a 'new identity' in Tor browser, a "[Javascript Application]" information box arises:  "Torbutton cannot safely give you a new identity.  It does not have access to the Tor Control Port.  Are you running Tor Browser Bundle?"

Reports:
[journalctl -f](http://pastebin.com/ME1aAQVS)
[/var/log/tor/log](http://pastebin.com/mKeejQjJ)

**Comment by dma** [comment]
Can you try:

sudo apt-get install roflcoptor

And then try rebooting?

**Comment by jlj2** [comment]
Very good suggestion.  Installing roflcoptor triggered a message over the desktop declaring 'Tor network successfully bootstrapped!'  Rebooting then demonstrated a 'green onion' button on the Tor browser with the tooltip 'Tor enabled'!

Notes:  
(1) In case this could be of interest, ```sudo apt-get install roflcoptor``` also involved apt-get installing procsnitchd concurrently (a dependency, presumably?)

(2) For what it's worth, the onion applet in the top Gnome bar is still greyed out and non-clickable, even where it appears in that Gnome bar's settings drop box with the name 'Tor Network'.

(3) The installed system (installed using a Subgraph iso from June 2016) had not yet gone through a full update when this issue was logged (and likewise issue #174 re backlight dropping off):  efforts to update were unsuccessful and would cause a system hang (see issue #177).  Perhaps relevant updates of certain Subgraph packages might have already resolved this current issue about Tor seeming to be disabled from install (on this system at least), or unless roflcoptor (and procsnitchd?) are installed also.  Immediately after trying your solution (roflcoptor + rebooting), a complete update was also finally successful (i.e. ```sudo apt update```, ```sudo apt upgrade```, and ```sudo apt dist-upgrade --auto-remove --purge```)!

**Comment by dma** [comment]
Hi, yes, roflcoptor and procsnitchd were added to one of the metapackages but aren't being delivered to users for a yet unknown reason. We're going to be making a new ISO soon enough that will include the more recently added new packages.

The roflcoptor update should fix the greyed out Tor icon. Please let me know if it doesn't. The frozen firewall dialog could be related: the fw prompt is a gnome-shell integration. Without roflcoptor installed, and post-update, the gnome shell may hang due to the Tor status extention failing to contact the Tor control port - Roflcoptor is a filter that was put in place to limit application access to the Tor control port.

Can you please confirm that after a reboot the top-bar onion is still greyed out?

**Comment by mckinney-subgraph** [comment]
There is not enough info for me to fix anything here. :( 

To better debug this issue, I need more info about what the system install state is like when the issue is occurring. The following command will print the installed versions of potentially related packages (and show any missing packages):
```
$ apt-cache policy subgraph-tor-config subgraph-standard roflcoptor procsnitchd tor
```

Also, it would be good to have this information at different stages:
1. While it is occurring and prior to "apt-get update" (no upgrade/dist-upgrade)
2. While it is occuring and after "apt-get update" (no upgrade/dist-upgrade)
3. After "apt-get update && apt-get upgrade"
4. After "apt-get update && apt-get dist-upgrade"

**Comment by mckinney-subgraph** [comment]
Also, there is another issue here that occurs after installing roflcoptor and procsnitchd -- a reboot shouldn't be needed to have the Tor control port working again. 

But there are a few things to consider:
1. The Tor status button may not refresh if it is not reloaded (try running: `alt-f2 + r`)
2. If the Tor browser is running when the packages are installed, it may need to be restarted to work with roflcoptor.

So if after applying steps 1 and 2, roflcoptor is not doing its job, then we will need to better capture why the roflcoptor service is not starting properly after being installed (but prior to rebooting).

**Comment by jlj2** [comment]
@dma : Many thanks for your detailed explanation.  The top-bar onion has still been greyed out during every session since the full update mentioned above, except perhaps it may have appeared lit at the time of the update:  I didn't consider it important to look out for it seeing how the green onion appears consistently now near the browser address bar, but towards the end of either that update session or the next session or two the Gnome top-bar onion was noticed to be lit, and the Tor Network entry in the drop down box was clickable.  I don't believe it has been on since.
Note:  I wonder whether my system may have been tainted since I decided to use about 5 addons.  Soon after installing one from a search in the addons sidebar, I was surprised to note that Tor's dropdown menus had a Xfce or LXDE look, yet during the current session the look has reverted to Gnome.  Notice also that the address bar and the active tab form a light grey shape like an upside-down capital 'T', and there are tinges of white to either side (see one side where the cursor points at), which doesn't seem to look normal:
![screenshot from 2016-09-01 22-33-44](https://cloud.githubusercontent.com/assets/17560937/18190955/eacd1362-70b5-11e6-9c91-1236d6af2c13.png)

**Comment by jlj2** [comment]
@ mckinney-subgraph:  
1. The green onion is showing in Tor browser consistently now, but the onion is greyed out in the Gnome top-bar, and no update has been done during the current session.  The result of ```apt-cache policy subgraph-tor-config subgraph-standard roflcoptor procsnitchd tor``` is [here](http://pastebin.com/9n5Hc4y6).

2. Green onion in browser, greyed out onion in top-bar, and after ```sudo apt-get update```, the result is [here](http://pastebin.com/pnX5maRa).

3. Immediately thereafter, did ```sudo apt-get update && sudo apt-get upgrade``` and the result is [here](http://pastebin.com/2RVdDV2N).

4. Then did ```sudo apt-get update && sudo apt-get dist-upgrade```, which offered new package ```libprotobuf10``` and an upgrade to ```libphonenumber7```, which were accepted ok.  The result is [here](http://pastebin.com/KQpstUSE).

**Comment by jlj2** [comment]
@ mckinney-subgraph : It is an important tip to do ```Alt-F2 + r``` because this now brings on a lit onion in the top Gnome bar.  This was attempted only after the update, upgrade and dist-upgrade described above.  The Gnome drop down menu also offers a 'New identity' under 'Tor Network'.  Thank you all!

**Comment by mckinney-subgraph** [comment]
So the issue of the Tor settings (`/etc/tor/torrc`) being updated before `roflcoptor` and `procsnitchd` packages are installed is now fixed. `apt-get upgrade` will no longer install Tor settings that require these packages. `apt-get dist-upgrade` must be executed to install the packages and the associated Tor settings. I will close this issue and we can create new issues related to the other problems if need be.

**Comment by jlj2** [comment]
In case it could be relevant for any related issue, please note that the ethernet cable was typically connected only once the Gnome login was completed.
